store/sdk/src/main/java/org/apache/carbondata/sdk/file/AvroCarbonWriter.java

```diff
@@ -213,6 +218,225 @@ private Object avroFieldToObject(Schema.Field avroField, Object fieldValue) {
         }
         out = new ArrayObject(arrayChildObjects);
         break;
+      case BYTES:
+        // DECIMAL type is defined in Avro as a BYTE type with the logicalType property
+        // set to "decimal" and a specified precision and scale
+        if (logicalType instanceof LogicalTypes.Decimal) {
+          out = new BigDecimal(new String(((ByteBuffer) fieldValue).array(),
+              CarbonCommonConstants.DEFAULT_CHARSET_CLASS));
+          ;
+        } else {
+          out = fieldValue;
+        }
+        break;
+      case UNION:
+        List<Schema> fieldsUnion = avroField.schema().getTypes();
+        int countIfNotNull = 0;
+        for (Schema unionField : fieldsUnion) {
+          if (!unionField.getType().equals(Schema.Type.NULL)) {
+            countIfNotNull++;
+          }
+        }
+        Object[] values;
+        values = new Object[countIfNotNull];
+        int j = 0;
+        for (Schema unionField : fieldsUnion) {
+          if (!unionField.getType().equals(Schema.Type.NULL)) {
+            values[j] = avroFieldToObjectForUnionType(unionField, fieldValue, avroField);
```

Comment:
1. Try to reuse the code as much as possible. You can extract the primitives types computation in a separate method and override only complex types as different methods for union and rest of the data types
2. Write a function for union type to identify the schema for which the computation need to be done. Do not call the computation for all the union types as at one time only one type of value will exists